golem-registry-service/db/migration/postgres/001_init.sql

```diff
@@ -650,3 +650,55 @@ CREATE TABLE environment_plugin_installation_revisions
 
 CREATE INDEX environment_plugin_installation_revisions_plugin_idx
     ON environment_plugin_installation_revisions (plugin_id);
+
+CREATE TABLE environment_shares
+(
+    environment_id  UUID      NOT NULL,
+    environment_share_id UUID NOT NULL,
+    grantee_account_id UUID NOT NULL,
+
+    created_at  TIMESTAMP NOT NULL,
+    updated_at  TIMESTAMP NOT NULL,
+    deleted_at      TIMESTAMP,
+    modified_by UUID      NOT NULL,
+
+    current_revision_id BIGINT    NOT NULL,
+
+    CONSTRAINT environment_shares_pk
+        PRIMARY KEY (environment_share_id),
+    CONSTRAINT environment_shares_environments_fk
+        FOREIGN KEY (environment_id) REFERENCES environments,
+    CONSTRAINT environment_shares_accounts_fk
+        FOREIGN KEY (grantee_account_id) REFERENCES accounts (account_id)
+);
+
+CREATE UNIQUE INDEX environment_shares_environment_grantee_uk
+    ON environment_shares (environment_id, grantee_account_id)
+    WHERE deleted_at IS NULL;
+
+CREATE TABLE environment_share_revisions
```

Comment:
for env component plugin installations i collapsed the revision - revision-plugin-relations tables into one. it is also works like this, but i think for these specfic versioned relations it is fine to skip the "empty" revision table

Reply:
I think it's nice to keep this structure as it's easier to add additional columns later on.
In the domain an environment share is very much it's own thing (considering we even have a ui view that lists all of the shares and lets people update / delete them), so I think it's worth keeping.